.projenrc.ts

```diff
@@ -1,12 +1,193 @@
-import { CdklabsConstructLibrary } from 'cdklabs-projen-project-types';
-const project = new CdklabsConstructLibrary({
-  author: 'AWS',
-  authorAddress: 'aws-cdk-dev@amazon.com',
-  cdkVersion: '2.1.0',
-  defaultReleaseBranch: 'main',
-  devDeps: ['cdklabs-projen-project-types'],
-  name: '@cdklabs/cdk-amazonmq-activemq-configs',
-  projenrcTs: true,
-  release: false,
+/*
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+import fs from "fs";
+import {
+  CdkLabsMonorepo,
+  TypeScriptWorkspace,
+} from "cdklabs-projen-project-types/lib/yarn";
+import { ReleasableCommits, SampleDir, TextFile } from "projen";
+import { Jsii } from "./projenrc/jsii";
+import { Stability } from "projen/lib/cdk";
+import { SourceUpdate } from "./projenrc/update-source";
+
+const stability = Stability.EXPERIMENTAL;
+
+const project = new CdkLabsMonorepo({
+  defaultReleaseBranch: "main",
+  stability,
+  deps: ["semver"],
+  devDeps: ["cdklabs-projen-project-types", "@types/semver"],
+  name: "@cdklabs/cdk-amazonmq-activemq-configs",
+  github: true,
+  release: true,
+  tsconfigDev: {
+    compilerOptions: {
+      lib: ["ES2021", "ES2021.String"]
+    }
+  }
 });
-project.synth();
\ No newline at end of file
+
+new SourceUpdate(project, {
+  name: 'config-xsd',
+  source: {
+    url: 'https://docs.aws.amazon.com/amazon-mq/latest/developer-guide/amazon-mq-broker-configuration-parameters.html',
+    xsdUrlPrefix: 'https://docs.aws.amazon.com/amazon-mq/latest/developer-guide/samples/'
+  },
+  cwd: 'sources'
+});
+
+project.gitignore.exclude("__MACOSX/");
+
+const xsd2jsii = new TypeScriptWorkspace({
+  parent: project,
+  private: true,
+  stability,
+  name: "@cdklabs/xsd2jsii",
+  packageName: "xsd2jsii",
+  bin: {
+    xsd2jsii: "bin/xsd2jsii",
+  },
+  deps: ["json5", "xml2js", "@faker-js/faker"],
+  bundledDeps: ["json5", "xml2js", "@faker-js/faker"],
+  devDeps: ["@types/xml2js", 'xsd-schema-validator'],
+  prettier: true,
+  tsconfigDev: {
+    include: ["bin/**/*.ts"],
+  },
+});
+
+xsd2jsii.addFields({ stability });
+
+const schemas = fs.readdirSync("./sources");
+
+const schemaReadmeTemplate = fs.readFileSync(
+  "./templates/README_cdk-amazonmq-activemq-config.TEMPLATE.md",
+  "utf-8"
+);
+
+const generatedTestTsTemplate = fs.readFileSync(
+  "./templates/generated.test.ts.template",
+  "utf-8"
+);
+
+const versionRegex = /^amazon-mq-active-mq-(\d+)\.(\d+)\.(\d+)\.xsd$/;
+
+for (const schema of schemas) {
+  const match = versionRegex.exec(schema);
+  if (!match) {
+    continue;
+  }
+
+  const [_, MAJOR, MINOR, PATCH] = match;
+
+  const schemaProject = new TypeScriptWorkspace({
+    parent: project,
+    stability,
+    name: `@cdklabs/cdk-amazonmq-activemq-config-v${MAJOR}-${MINOR}-${PATCH}`,
```

Comment:
For my understanding, this major, minor, patch refers to the Spring XML schema version right? So in the future, in a specific package release version, it can theoretically contains different XML schema versions in the `packages` dir right?

IMO, I think that's fine and this approach makes sense to me.

Reply:
That's right. It is distinct from the _package release versions_ that will be changing as we evolve the library. That is why this is part of the names.